source/game/system/map/MapdataCheckPoint.hh

```diff
@@ -20,27 +30,57 @@ public:
 
     MapdataCheckPoint(const SData *data);
     void read(EGG::Stream &stream);
+    void initCheckpointLinks(MapdataCheckPointAccessor &accessor, int id);
+
+    /// @beginGetters
+    [[nodiscard]] s8 checkArea() const;
+    /// @endGetters
+
+    static constexpr s8 NORMAL_CHECKPOINT = -1; ///< Only used for picking respawn position
+    static constexpr s8 FINISH_LINE = 0;        ///< Triggers a lap change
```

Comment:
Wouldn't this fit better as an enum class?

Reply:
wouldn't it have to be a regular enum? im not a cpp expert, but it doesnt like it as an enum class because `checkArea` is an s8.

Reply:
> static cast to the enum class's underlying type

oh ofc, okay got it 👍